server/errorHandler.ts

```diff
@@ -20,9 +21,34 @@ export default function createErrorHandler(production: boolean) {
       if (err.status === 403 && (!err.external || err.response?.body?.accessErrors)) {
         res.status(403)
 
+        const userHeaderService = 'You do not have permission to view this service'
+        const userHeaderPage = 'You do not have permission to view this page'
+        const returnedError = err.response?.body?.message
+        let userMessage
+        let userHeader
+
+        if (returnedError === 'could not map service provider user to access scope') {
+          userMessage = errorMessages.errorHandlerAccessErrorMessages.notSetUpCorrectly
+          userHeader = userHeaderService
+        } else if (returnedError === 'user is not a service provider') {
+          userMessage = errorMessages.errorHandlerAccessErrorMessages.notSetUpCorrectly
+          userHeader = userHeaderPage
+        } else if (returnedError === 'cannot find user in hmpps auth') {
+          userMessage = errorMessages.errorHandlerAccessErrorMessages.emailNotRecognised
+          userHeader = userHeaderPage
```

Comment:
adding error message will involve adding more and more `else if` checks, could we not move this to a lookup ie in `errorMessages.ts`

```
const errorHandlerAccessErrorMessages = {
  notSetUpCorrectly:
    'Your account is not set up correctly. Ask an admin user in your organisation to fix this in HMPPS Digital Services.',
  emailNotRecognised:
    'Your email address is not recognised. If it has changed recently, try signing out and signing in with the correct one. Ask an admin user in your organisation to check what the right email is in HMPPS Digital Services. If that does not work, <a target="_blank" href="/report-a-problem">report it as a problem.</a>',
  providerGroupNotRecognised:
    'Your provider group is not recognised. Ask an admin in your organisation to check it has been set up correctly in HMPPS Digital Services. They may need to <a target="_blank" href="/report-a-problem">report it as a problem.</a>',
  contractGroupNotRecognised:
    'Your contract group is not recognised. Ask an admin in your organisation to check it has been set up correctly in HMPPS Digital Services. They may need to <a target="_blank" href="/report-a-problem">report it as a problem.</a>',
}
const userHeaderTypes = {
  userHeaderService: 'You do not have permission to view this service',
  userHeaderPage: 'You do not have permission to view this page',
  default: 'Sorry, you are not authorised to access this page',
}
const returnedError = {
  'could not map service provider user to access scope': {
    mappedMessage: errorHandlerAccessErrorMessages.notSetUpCorrectly,
    userHeaderType: userHeaderTypes.userHeaderService,
  },
}
```
(then export these variables)

and change `errorHandler.ts` :

```
        const userHeaderTypes = {
          userHeaderService: 'You do not have permission to view this service',
          userHeaderPage: 'You do not have permission to view this page',
        }

        const lookupError = errorMessages.returnedError[returnedError]
        if (lookupError) {
          userMessage = lookupError.mappedMessage
          userHeader = userHeaderTypes[lookupError.userHeaderType]
        }
```
(and lookup the other userHeader types from userMessages)

doesn't help with the `.includes` errors but should make it easier to add new errors and removes hard coded messages from `errorHandler.ts`